src/utils/pluginDriver.ts

```diff
@@ -39,22 +41,36 @@ export type HookContext = (context: PluginContext, plugin?: Plugin) => PluginCon
 export function createPluginDriver(
 	graph: Graph,
 	options: InputOptions,
+	pluginCache: Record<string, SerialisablePluginCache>,
 	watcher?: Watcher
 ): PluginDriver {
 	const plugins = [...(options.plugins || []), getRollupDefaultPlugin(options)];
 	const { emitAsset, getAssetFileName, setAssetSource } = createAssetPluginHooks(graph.assetsById);
+	const existingPluginKeys: string[] = [];
 
 	let hasLoadersOrTransforms = false;
 
 	const pluginContexts = plugins.map(plugin => {
+		let cacheable = true;
+		if (typeof plugin.name === 'string') {
+			if (existingPluginKeys.indexOf(plugin.name) === -1) cacheable = false;
```

Comment:
Is this correct? Do you mean `!== -1` here? I'm a little confused at this block of code.

Reply:
Yup, did not mean to imply this was bug-free just yet :)